recommender/recommender.py

```diff
@@ -657,43 +678,106 @@ def endorse_resource(self, data, _suffix=''):
         result['recommendation'] = deendorsed_resource
         tracker.emit('deendorse_resource', result)
         return result
 
     @XBlock.json_handler
+    def export_resources(self, _data, _suffix):
+        """
+        Export all resources from the Recommender. This is intentionally not limited to staff
+        members (community contributions do not belong to the course staff). Sensitive
+        information is exported *is* limited (flagged resources, and in the future, PII if
+        any).
+        """
+        result = {}
+        result['export'] = {
+            'recommendations': self.recommendations,
+            'deendorsed_recommendations': self.deendorsed_recommendations,
+            'endorsed_recommendation_ids': self.endorsed_recommendation_ids,
+            'endorsed_recommendation_reasons': self.endorsed_recommendation_reasons,
+        }
+        if self.get_user_is_staff():
+            result['export']['flagged_accum_resources'] = self.flagged_accum_resources
+
+        result['Success'] = True
+        tracker.emit('export_resources', result)
+        return result
+
+    @XBlock.handler
+    def import_resources(self, request, _suffix=''):
+        """
+        Import resources into the recommender.
+        """
+        response = Response()
+        response.headers['Content-Type'] = 'text/plain'
+        if not self.get_user_is_staff():
+            response.body = 'NOT_A_STAFF'
+            tracker.emit('import_resources', {'Status': response.body})
+            return response
+
+        raw_data = ''
+        try:
+            with request.POST['file'].file as lines:
+                for line in lines:
+                    raw_data += ' ' + line
+
+            raw_data = raw_data.strip()
+            data = json.loads(raw_data)
+
+            self.flagged_accum_resources = data['flagged_accum_resources']
+            self.endorsed_recommendation_reasons = data['endorsed_recommendation_reasons']
+            self.endorsed_recommendation_ids = data['endorsed_recommendation_ids']
+
+            if 'deendorsed_recommendations' in data:
+                self.deendorsed_recommendations = data_structure_upgrade(data['deendorsed_recommendations'])
+                data['deendorsed_recommendations'] = self.deendorsed_recommendations
+            self.recommendations = data_structure_upgrade(data['recommendations'])
+            data['recommendations'] = self.recommendations
+
+            tracker.emit('import_resources', {'Status': 'SUCCESS', 'data': data})
+            response.body = json.dumps(data, sort_keys=True)
+            return response
+        except:
+            response.body = 'FILE_FORMAT_ERROR'
+            tracker.emit('import_resources', {'Status': response.body, 'data': raw_data})
+            return response
+
+    @XBlock.json_handler
     def get_accum_flagged_resource(self, _data, _suffix=''):
+        """
+        Accumulate the flagged resource ids and reasons from all students
```

Comment:
The docstring uses the verb `accumulate`. Why isn't the function just called `accum_flagged_resource`? Why is the `get_` useful?